controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java

```diff
@@ -517,6 +545,141 @@ private SlimeJsonResponse tenantInfo(TenantInfo info, HttpRequest request) {
         return new SlimeJsonResponse(slime);
     }
 
+    private SlimeJsonResponse tenantInfoProfile(CloudTenant cloudTenant) {
+        var slime = new Slime();
+        var root = slime.setObject();
+        var info = cloudTenant.info();
+
+        if (!info.isEmpty()) {
+            var contact = root.setObject("contact");
+            contact.setString("name", info.contact().name());
+            contact.setString("email", info.contact().email());
+
+            var tenant = root.setObject("tenant");
+            tenant.setString("company", info.name());
+            tenant.setString("website", info.website());
+
+            toSlime(info.address(), root); // will create "address" on the parent
+        }
+
+        return new SlimeJsonResponse(slime);
+    }
+
+    private SlimeJsonResponse putTenantInfo(String tenantName, HttpRequest request, BiFunction<CloudTenant, Inspector, SlimeJsonResponse> handler) {
+        return controller.tenants().get(tenantName)
+                .map(tenant -> handler.apply((CloudTenant) tenant, toSlime(request.getData()).get()))
+                .orElseGet(() -> ErrorResponse.notFoundError("Tenant '" + tenantName + "' does not exist or does not support this"));
+    }
+
+    private SlimeJsonResponse putTenantInfoProfile(CloudTenant cloudTenant, Inspector inspector) {
+        var info = cloudTenant.info();
+
+        var mergedContact = TenantContact.empty()
+                .withName(getString(inspector.field("contact").field("name"), info.contact().name()))
+                .withEmail(getString(inspector.field("contact").field("email"), info.contact().email()));
+
+        var mergedAddress = updateTenantInfoAddress(inspector.field("address"), info.address());
+
+        var mergedInfo = info
+                .withName(getString(inspector.field("tenant").field("name"), info.name()))
+                .withWebsite(getString(inspector.field("tenant").field("website"), info.website()))
+                .withContact(mergedContact)
+                .withAddress(mergedAddress);
+
+        validateMergedTenantInfo(mergedInfo);
+
+        controller.tenants().lockOrThrow(cloudTenant.name(), LockedTenant.Cloud.class, lockedTenant -> {
+            lockedTenant = lockedTenant.withInfo(mergedInfo);
+            controller.tenants().store(lockedTenant);
+        });
+
+        return new MessageResponse("Tenant info updated");
```

Comment:
Maybe this should be part of the common function? Why isn't `validateMergedTenantInfo()` called in the other `put*` methods?

Reply:
Because it does not validate any of the information that is part of this update. We should probably split this into two different validation methods because part of the issue we are trying to solve here is that I can't update a tenant that has bad information in both console forms :)